src/form.js

```diff
@@ -208,6 +208,10 @@ export default class SimpleteForm extends HTMLElement {
 		return !this.searchField.value.trim();
 	}
 
+	get navigating() {
+		return this.searchField.hasAttribute("aria-activedescendant");
```

Comment:
Given that this is invoked on every keypress, we might want to employ memoization here? That is, for performance reasons I believe we want to avoid querying the DOM every time and thus need an instance property after all.

Reply:
I'm not opposed to doing that and will look into it, but I think I do not fully understand the previous implementation (e.g. I don't fully understand why `this.navigating` is set only when the `preview` variable is true).

Reply:
Frankly, I don't really understand that part anymore either. In fact, I'm not sure I ever did (cf. b426bc3a41f909e295333887e96126b7a2651b2b) - which is why I periodically feel/suppress the urge to rewrite everything around a proper state machine...

Reply:
Upon review of the code, it seems that `preview` referred to the act of
viewing a suggestion without actually selecting it (e.g. it controlled
whether or not the dropdown window would close upon selection).

This is exactly the meaning of `navigating` in the `SimpleteForm`
component, so e0144d5b4c8a619f8d9a29603d32476ac9242078 refactors the `SimpleteSuggestions` component to use
a property named `navigating` instead of `preview`.

e0144d5b4c8a619f8d9a29603d32476ac9242078 also restructures the code to use memoization of the navigating
state for performance reasons.